### PR TITLE
*: fix conflict of the ddl seq number  

### LINKED ISSUE #32402: support session variable `tidb_last_ddl_info`
## Feature Request

**Is your feature request related to a problem? Please describe:**

Since TiDB already support `tidb_last_txn_info`, it's reasonable to support `tidb_last_ddl_info`.
The client may want to know the order of the concurrent DDLs. We want to give this order info in `tidb_last_ddl_info`.
For example, we have two DDLs, namely DDL1 and DDL2, sent by two sessions concurrently.
We can't know which DDL is executed first.
After introducing `tidb_last_ddl_info`, we can use `select @@tidb_last_ddl_info` to get `x` for the first executed DDL and `x+1` for the second executed DDL.

### PR DESCRIPTION
**Pull request by wjhuang2016**
### What problem does this PR solve?

Issue Number: close #32402
A patch for #32414
Problem Summary:
Concurrent workers call finishDDLJob may cause the DDL seq num conflict.

### What is changed and how it works?
Use mutex to avoid it.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
The test is covered by schrddl. PR: https://github.com/PingCAP-QE/schrddl/pull/17
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: ddl/ddl_worker.go [hunk 1]
```diff
@@ -440,34 +444,16 @@ func (w *worker) finishDDLJob(t *meta.Meta, job *model.Job) (err error) {
...
-		job.SeqNum, err = t.GetHistoryDDLCount()
-		// Make it start from 1.
-		job.SeqNum += 1
-	}
-	if err != nil {
-		return err
-	}
-	return nil
+func (w *worker) writeDDLSeqNum(job *model.Job) {
+	ddlSeqNumMu.Lock()
```
Comment (hawkingrei) [inline ddl/ddl_worker.go:453]: Is there a way to make Lock/Unlock of ddlSeqNumMu together?

Comment (wjhuang2016) [reply ddl/ddl_worker.go:453]: Maybe not

**Review by hawkingrei — APPROVED** [review]

**Review by xiongjiwei — APPROVED** [review]

**Pull request merged**